vars/kolaTestIso.groovy

```diff
@@ -1,102 +1,9 @@
-    def ids = []
-
-    // If given a marker then add it to the parallel run stage titles
-    def titleMarker = marker == "" ? "" : "${marker}:"
-
-    def testIsoRuns = [:]
-    testIsoRuns["${titleMarker}kola:metal"] = {
-        def id = marker == "" ? "kola-testiso-metal" : "kola-testiso-metal-${marker}"
-        ids += id
-        def scenariosArg = scenarios == "" ? "" : "--scenarios ${scenarios}"
-        shwrap("cd ${cosaDir} && cosa kola testiso -S ${extraArgs} ${scenariosArg} --output-dir ${outputDir}/${id}")
-    }
-    if (!params['skipMetal4k']) {
-        // metal4k test doesn't work on s390x for now
-        // https://github.com/coreos/fedora-coreos-tracker/issues/1261
-        // and testiso for s390x doesn't support iso installs either
-        if (arch != 's390x') {
-            testIsoRuns["${titleMarker}kola:metal4k"] = {
-                def id = marker == "" ? "kola-testiso-metal4k" : "kola-testiso-metal4k-${marker}"
-                ids += id
-                def scenariosArg = scenarios4k == "" ? "" : "--scenarios ${scenarios4k}"
-                shwrap("cd ${cosaDir} && cosa kola testiso -S --qemu-native-4k ${extraArgs4k} ${scenariosArg} --output-dir ${outputDir}/${id}")
-            }
-        }
-    }
-    if (!params['skipMultipath']) {
-        testIsoRuns["${titleMarker}kola:multipath"] = {
-            def id = marker == "" ? "kola-testiso-multipath" : "kola-testiso-multipath-${marker}"
-            ids += id
-            shwrap("cd ${cosaDir} && cosa kola testiso -S --qemu-multipath ${extraArgsMultipath} --scenarios ${scenariosMultipath} --output-dir ${outputDir}/${id}")
-        }
-    }
-    if (!params['skipUEFI']) {
-        // only aarch64 and x86_64 support UEFI. aarch64 UEFI was
-        // already tested in the basic run above and secureboot isn't
-        // supported in aarch64 so we limit this to x86_64 for now.
-        // https://pagure.io/fedora-infrastructure/issue/7361
-        // https://github.com/coreos/coreos-assembler/blob/93efb63dcbd63dc04a782e2c6c617ae0cd4a51c8/mantle/platform/qemu.go#L1156
-        if (arch == 'x86_64') {
-            testIsoRuns["${titleMarker}kola:uefi"] = {
-                def id = marker == "" ? "kola-testiso-uefi" : "kola-testiso-uefi-${marker}"
-                ids += id
-                shwrap("cd ${cosaDir} && cosa shell -- mkdir -p ${outputDir}/${id}")
-                shwrap("cd ${cosaDir} && cosa kola testiso -S --qemu-firmware=uefi ${extraArgsUEFI} --scenarios ${scenariosUEFI} --output-dir ${outputDir}/${id}/insecure")
-                if (!params['skipSecureBoot']) {
-                    shwrap("cd ${cosaDir} && cosa kola testiso -S --qemu-firmware=uefi-secure ${extraArgsUEFI} --scenarios ${scenariosUEFI} --output-dir ${outputDir}/${id}/secure")
-                }
-            }
-        }
-    }
-
-    try {
-        // Run at most two testiso runs at a time to try not to
-        // exceed 8G of memory usage.
-        utils.runParallel(testIsoRuns, 2)
```

Comment:
This is an interesting consideration. `kola testiso` doesn't support parallel execution right now, but we've hacked around this here by running it in parallel. We lose that with this work. I think that's OK, but just noting it since it'll make the pipeline a bit slower. Long-term, this will be fixed when we merge into `kola run`. Though shorter-term, we could support something similar to https://github.com/coreos/coreos-assembler/pull/3272 in `kola testiso` to reintroduce some parallelism here. But let's not block on that!